cfgov/unprocessed/js/apps/financial-well-being/fwb-questions.js

```diff
@@ -1,83 +1,169 @@
-/* eslint no-global-assign: "off" */
 'use strict';
 
-var Analytics = require( '../../modules/Analytics' );
+const Analytics = require( '../../modules/Analytics' );
 
+/**
+ * Initialize the questionnaire
+ */
 function init() {
+  const questionElems = document.querySelectorAll( '#quiz-form fieldset' );
+  const radioButtons = document.querySelectorAll( '#quiz-form [type="radio"]' );
+  const submitButton = document.querySelector( '#submit-quiz' );
+  const questionStates = {};
+
+  /**
+   * Checks the selection status of a group of radio inputs,
+   * belonging to a parent fieldset
+   * @param {HTMLNode} questionRadios An array of dom elements.
+   * @returns {boolean} The status of the group of inputs
+   */
+  function checkQuestionState( questionRadios ) {
+    let radioIsChecked = false;
+
+    for ( let i = 0, l = questionRadios.length; i < l; i++ ) {
```

Comment:
Yep, for the `i < l` condition

Reply:
I mean, why not just do `i < questionRadios.length`? I should probably know the answer to this, but I can't remember. Thanks for indulging me :)

Reply:
Each loop you are re-calculating the length of `questionRadios`. By setting the length to a variable you're storing the length and just referencing it every loop.

Reply:
Please change the  `l = questionRadios.length` to `questionRadiosLength`. It's more verbose but it makes the code easier to read.

Reply:
@anselmbradford Ah, right. I think I keep forgetting this because it's obviously a poor implementation decision that this is the case :trollface: (JK; I'm sure there's a reason it's not stupid.)

Reply:
@Scotchester it's not so much implementation decision as that's just how a lookup works. If you were to do a lookup throughout a loop in any language you'd have the same thing happening. Think of it like saving a DOM query to a variable vs looking it up every time you need it.

Reply:
@jimmynotjim Yes, I get that part :P I was thinking it should somehow be smart enough to cache the non-counter pieces of the expression, but I suppose it's possible you want to change that within the code that the loops executes.

Reply:
I wouldn't want to see it make that kind of assumption, especially if items were added to the object within the loop.

Reply:
@sebworks are we sure this is easier to read? https://github.com/cfpb/cfgov-refresh/pull/3417/commits/c0b85a048f580f824cc6bbbc3db62af1cb19528a

Reply:
lol, it's gigantic, try `qLen` and `rLen`.

Reply:
@sebworks how bout this? https://github.com/cfpb/cfgov-refresh/pull/3417/commits/2b40d3974fbd470a19d152eeb995ca76d1d13428 Even with a shorter variable name, it was still longer than 80 chars

Reply:
yep, that is money.